hub/api/dataset.py

```diff
@@ -187,15 +187,37 @@ def delete(path: str, force: bool = False, large_ok: bool = False) -> None:
 
     @staticmethod
     def like(
-        path: str, like: str, like_creds: dict, overwrite: bool = False
+        path: str,
+        source: Union[str, Dataset],
+        creds: dict = None,
+        overwrite: bool = False,
     ) -> Dataset:
-        """Creates a dataset with the same structure as another dataset"""
-        raise NotImplementedError
+        """Copies the `source` dataset's structure to a new location. No samples are copied, only the meta/info for the dataset and it's tensors.
+
+        Args:
+            path (str): Path where the new dataset will be created.
+            source (Union[str, Dataset]): Path or dataset object that will be used as the template for the new dataset.
+            creds (dict): Credentials that will be used to create the new dataset.
+            overwrite (bool): If True and a dataset exists at `destination`, it will be overwritten. Defaults to False.
+
+        Returns:
+            Dataset: New dataset object.
+        """
+
+        destination_ds = dataset.empty(path, creds=creds, overwrite=overwrite)
+        source_ds = source
+        if isinstance(source, str):
+            source_ds = dataset.load(source)
+
+        for tensor_name in source_ds.meta.tensors:  # type: ignore
```

Comment:
Is this failing because `source_ds` can be string-type? You can probably annotate it explicitly